src/zenml/services/local/local_service.py

```diff
@@ -167,20 +167,21 @@ def pid(self) -> Optional[int]:
                     return None
 
                 config_file = self.config_file
-                if config_file is None:
-                    return pid
                 if (
-                    daemon_entrypoint.__name__ not in cmd_line
-                    or config_file not in cmd_line
-                ) and (
-                    daemon_entrypoint.__name__ not in cmd_line[0]
-                    or config_file not in cmd_line[0]
+                    config_file is not None
+                    and (
+                        daemon_entrypoint.__name__ not in cmd_line
+                        or config_file not in cmd_line
+                    )
+                    and (
+                        daemon_entrypoint.__name__ not in cmd_line[0]
+                        or config_file not in cmd_line[0]
+                    )
                 ):
-                    logger.debug(
+                    logger.warning(
                         f"Process with PID {pid} is not a ZenML local daemon "
                         f"service."
                     )
-                    return None
```

Comment:
Does this change impact other services built on top of `local_service`?

Reply:
Not sure either, I implemented it how I understood @stefannica's review comment on #1604. @stefannica do you think it makes sense how it is implemented now? Also, do you have an answer to @safoinme's question?

Reply:
This is an extra validation to ensure that the PID of a ZenML daemon process is indeed a daemon process and not some other process that happened to reuse the PID of a ZenML daemon process that was not properly cleaned up. It doesn't work in bentoml's case, so we can just make it a warning.